database/raft/raft.go

```diff
@@ -208,23 +222,10 @@ func Start(laddr, dir, bootURL string, httpClient *http.Client, useTLS bool) (*S
 		Logger:          &raft.DefaultLogger{Logger: stdlog.New(ioutil.Discard, "", 0)},
 	}
 
-	if walobj != nil {
+	if recover {
 		sv.raftNode = raft.RestartNode(c)
 		triggerElection(ctx, sv)
```

Comment:
I updated the patch to preserve the existing behavior, so we do this when recovering from an existing WAL but not when joining.